src/actions/reports.ts

```diff
@@ -0,0 +1,23 @@
+'use server'
+import { reportDeal } from '@/queries/reports'
+import { reportSchema } from '@/schemas/reports'
+import { ReturnValue } from '@/types/Types'
+
+export const reportDealAction = async (
+  dealId: string,
+  reason: string,
+  email: string
+): Promise<ReturnValue<undefined>> => {
```

Comment:
What is instead of having this as `Promise<ReturnValue<undefined>>`  you made an interface for the ReturnValue and made it more type safe? Or is that overkill/unnecessary?

```ts
interface ReportDealReturnValue {
  message: string;
  success: boolean;
}

export const reportDealAction = async (
  dealId: string,
  reason: string,
  email: string
): Promise<ReturnValue<ReportDealReturnValue>> => {
  ...
}
```

Reply:
The `ReturnValue` interface already includes those properties so they should already be taken care of. This may look a little off since we are using a generic for the return data type, but the `data` property is optional (not all functions need to return a data object). That's why I'm using `undefined` as the generic type although I agree it feels a little odd.

```typescript
export type ReturnValue<T> = {
  data?: T
  message?: string
  success: boolean
  error?: Error
}
```

Had you seen the full `ReturnValue` definition? Does this change anything?

Reply:
Ahah, this makes perfect sense and should have looked at the entire ReturnValue. I'll go ahead and approve this one.